src/rgw/rgw_common.h

```diff
@@ -1917,6 +1934,7 @@ struct rgw_obj {
   rgw_obj_key key;
 
   bool in_extra_data{false}; /* in-memory only member, does not serialize */
+  bool head_obj{true}; /* in-memory only member, does not serialize */
```

Comment:
@mikulely I actually prefer the explicitness of head_obj, and don't mind the minimal cost of computing it early.